public/app/features/alerting/unified/components/rule-editor/AnnotationsField.tsx

```diff
@@ -81,17 +128,68 @@ const AnnotationsField = () => {
             </div>
           );
         })}
-        <Button
-          className={styles.addAnnotationsButton}
-          icon="plus-circle"
-          type="button"
-          variant="secondary"
-          onClick={() => {
-            append({ key: '', value: '' });
-          }}
+        <Stack direction="row" gap={1}>
+          <Button
+            className={styles.addAnnotationsButton}
+            icon="plus-circle"
+            type="button"
+            variant="secondary"
+            onClick={() => {
+              append({ key: '', value: '' });
+            }}
+          >
+            Add new annotation
+          </Button>
+          <Button type="button" variant="secondary" icon="dashboard" onClick={setShowPanelSelector}>
+            Set panel and dashboard
```

Comment:
Nitpick, to be consistent with the order of words within the modal

```suggestion
            Set dashboard and panel
```